### PR TITLE
Issue-859/CI-6320 | Fixing Docker tests; Updating Docker software versions

### PR DESCRIPTION
**Pull request by mngi-arogers**
### Your checklist for this pull request
Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.

🚨Please review the [guidelines for contributing](../CONTRIBUTING.md) to this repository.

- [x ] Make sure you are making a pull request against the **develop branch** (left side). Also you should start *your branch* off *our develop*.
- [ x] Make sure you are requesting to pull request from a **topic/feature/bugfix branch** (right side). Don't pull request from your master!

What does this implement/fix? Explain your changes.
---------------------------------------------------
* This change ensures the Docker build installs PHP libraries according to `composer.lock`. This will fix https://github.com/wp-graphql/wp-graphql/issues/859
* Travis runs tests based on WordPress 5.2.0
* Docker
   * Updating PHP Composer
   * Updating MariaDB version
   * Setting default WordPress version to 5.2.0
   * Exposing MySQL port so devs can connect to Docker MySQL container with a MySQL client
 * Other
   * Added README instructions for using MySQL client on Docker setup

Does this close any currently open issues?
------------------------------------------
Yes, https://github.com/wp-graphql/wp-graphql/issues/859

Any relevant logs, error output, GraphiQL screenshots, etc?
-------------------------------------
N/A

Any other comments?
-------------------
N/A

Where has this been tested?
---------------------------
**Operating System:**
OS X: Mojave + Docker for Mac

**WordPress Version:**
5.2.0

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -23,9 +23,9 @@ env:
   matrix:
     # These apply to the latest WP version that is available as a Docker image: https://hub.docker.com/_/wordpress/
     - WP_VERSION=5.0.2 PHP_VERSION=5.6
-    - WP_VERSION=5.1.1 PHP_VERSION=7.1
-    - WP_VERSION=5.1.1 PHP_VERSION=7.2
-    - WP_VERSION=5.1.1 PHP_VERSION=7.3 COVERAGE=true LINT_SCHEMA=true
+    - WP_VERSION=5.2.0 PHP_VERSION=7.1
+    - WP_VERSION=5.2.0 PHP_VERSION=7.2
+    - WP_VERSION=5.2.0 PHP_VERSION=7.3 COVERAGE=true LINT_SCHEMA=true
```
Comment (jasonbahl) [inline .travis.yml]: 👌

**Comment by mngi-arogers** [comment]
> Looks good to me. Thanks for the update @mngi-arogers
> 
> What would it take to update to use the newer version of Codeception you mentioned in the issue?

The `composer.lock` file would need to be updated with the desired versions and then someone would have to investigate and address the PHP errors due to the new Codeception version. The Docker stuff would likely not need to be touched. Just the PHP code.

**Comment by jasonbahl** [comment]
@mngi-arogers 
> The composer.lock file would need to be updated with the desired versions and then someone would have to investigate and address the PHP errors due to the new Codeception version. The Docker stuff would likely not need to be touched. Just the PHP code.

Ok. We can address that update separately.

**Comment by mngi-arogers** [comment]
Hmm, getting an error when using PHP 5.6: https://travis-ci.org/wp-graphql/wp-graphql/jobs/534979256#L895

**Comment by mngi-arogers** [comment]
@jasonbahl 
Looks like the tests are failing with PHP 5.6 because `composer.lock` contains dependencies that require PHP 7.X or newer be used. Here's one example:
```
Your requirements could not be resolved to an installable set of packages.

  Problem 1
    - Installation request for doctrine/inflector v1.3.0 -> satisfiable by doctrine/inflector[v1.3.0].
    - doctrine/inflector v1.3.0 requires php ^7.1 -> your PHP version (5.6.39) does not satisfy that requirement.
```

Since `composer install` is the proper command to run on CI or production systems:
* https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies
* https://stackoverflow.com/a/33052263

...it looks like the `composer.lock` file needs to modified so it can work with PHP 5.6 and 7.X. As a temporary workaround, I've disabled tests on PHP 5.6 at least to make Travis less broken and make it possible for merges to have their PHP 7.X tests run for now.

**Pull request merged**